TrafficCapture/captureOffloader/src/main/java/org/opensearch/migrations/trafficcapture/StreamChannelConnectionCaptureSerializer.java

```diff
@@ -332,6 +342,9 @@ private void addSubstreamMessage(int captureFieldNumber, int dataFieldNumber, in
         }
         // Write data field
         writeByteBufferToCurrentStream(dataFieldNumber, byteBuffer);
+        if (captureFieldNumber == TrafficObservation.READ_FIELD_NUMBER) {
```

Comment:
One concern about the placement here. This should work fine for non-segmented messages as far as I can tell. For segmented messages it may be obvious for us from the Replayer that this is a continuation but at least here we are not setting this `readObservationsAreWaitingForEom = true`. Is this something we are checking in the Replayer?

Reply:
This can be ignored, even segmented messages enter this method

Reply:
Separately from our discussion, we will probably need this if condition to have `TrafficObservation.READSEGMENT_FIELD_NUMBER` for segmented requests that enter here

Reply:
Ahhh - here’s what I was thinking, though I think that I was wrong (I’ll write some tests).
If we get an observation that starts with a segment, we’re here because the message wasn’t big enough to fit within the available buffer.  I was thinking that that meant the PREVIOUS buffer, so that any traffic stream that started with a segment must mean that there were read bytes on the previous trafficStream.  However, we could have an observation bigger than our max TrafficStream size (10K message, 1K max size).  In that case, we’d need to segment the first message too - so we’d end up back in an ambiguous position.

Reply:
Assuming we will cover this in a update somewhere else

Reply:
I'm working on this within https://opensearch.atlassian.net/browse/MIGRATIONS-1379.